lighthouse-core/lib/icons.js

```diff
@@ -24,17 +24,26 @@ function doExist(manifest) {
  * @param {NonNullable<LH.Artifacts.Manifest['value']>} manifest
  * @return {Array<string>} Value of satisfactory sizes (eg. ['192x192', '256x256'])
  */
-function sizeAtLeast(sizeRequirement, manifest) {
+function pngSizedAtLeast(sizeRequirement, manifest) {
   // An icon can be provided for a single size, or for multiple sizes.
   // To handle both, we flatten all found sizes into a single array.
   const iconValues = manifest.icons.value;
   /** @type {Array<string>} */
   const flattenedSizes = [];
-  iconValues.forEach(icon => {
-    if (icon.value.sizes.value) {
-      flattenedSizes.push(...icon.value.sizes.value);
-    }
-  });
+  iconValues
+    // filter out icons with a typehint that is not 'image/png'
+    .filter(icon => !icon.value.type.value ||
+      (icon.value.type.value &&
+      icon.value.type.value === 'image/png'))
+    // filter out icons that are not png
+    .filter(icon => icon.value.src.value &&
+      icon.value.src.value.endsWith('.png'))
```

Comment:
some people add query strings in their manifest (e.g. `https://example.com/img.png?utm_source=manifest` or whatever for analytics), and the spec [doesn't disallow it](https://www.w3.org/TR/appmanifest/#src-member), so I think we want to do `new URL(icon.value.src.value).pathname.endsWith('.png')` here.

(note that `URL` needs to come from `const URL = require('./url-shim.js')`, which gets a different version of the constructor based on if it's running in node or in the browser)

Reply:
Good point, fixed!